File: Python praktikum/prak4/Soal4.py
#from array import *
data = [0, 0, 0, 0, 0, 0, 0, 0]

tanggal = int(input("Masukan Tanggal = "))

if tanggal <= 31:
    data[0] = float(tanggal)//10
    data[1] = tanggal%10
else:
    print("di kalender Hanya ada 31 hari")
    

bulan = str(input("Masukan Bulan = "))

if bulan == "Januari":
        data[2] = 0
        data[3] = 1
elif bulan == "Februari":
        data[2] = 0
        data[3] = 2
elif bulan == "Maret":
        data[2] = 0
        data[3] = 3
elif bulan == "April":
        data[2] = 0
        data[3] = 4
elif bulan == "Mei":
        data[2] = 0
        data[3] = 5
elif bulan == "Juni":
        data[2] = 0
        data[3] = 6
elif bulan == "Juli":
        data[2] = 0
        data[3] = 7
elif bulan == "Agustus":
        data[2] = 0
        data[3] = 8
elif bulan == "September":
        data[2] = 0
        data[3] = 9
elif bulan == "Oktober":
        data[2] = 1
        data[3] = 0
elif bulan == "November":
        data[2] = 1
        data[3] = 1
else:
    data[2] = 1
    data[3] = 2
    
tahun = int(input("Masukan Tahun = "))

for i in range(7, 3, -1):
    data[i] = tahun % 10
    tahun = float(tahun )// 10


j = 7
temp = 0
for i in range(0, 8, 1):
    if data[i] == data[j]:
        temp = temp
    else:
        temp += 1
    j -= 1
    

    

if temp == 0:
    print("Palindrome")
else:
    print("Not Palindrome")

    

File: java praktikum/prak11/JavaApplication19/src/javaapplication19/Soal4.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package javaapplication19;
import java.util.Scanner;
/**
 *
 * @author Nrhys
 */
public class Soal4 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int data[], tanggal, tahun, temp = 0, j;
        data = new int[8];
        String bulan;
        
        System.out.println("Inputkan Tanggal = ");
        tanggal = input.nextInt();
        
        if(tanggal <= 31){
            data[0] = tanggal/10;
            data[1] = tanggal%10;
        }else{
            System.out.println("di kalender Hanya ada 31 hari");
            return;
        }
        
        System.out.println("Inputkan Bulan = ");
        bulan = input.next();
        
        switch (bulan) {
            case "Januari" -> {
                data[2] = 0;
                data[3] = 1;
            }
            case "Februari" -> {
                data[2] = 0;
                data[3] = 2;
            }
            case "Maret" -> {
                data[2] = 0;
                data[3] = 3;
            }
            case "April" -> {
                data[2] = 0;
                data[3] = 4;
            }
            case "Mei" -> {
                data[2] = 0;
                data[3] = 5;
            }
            case "Juni" -> {
                data[2] = 0;
                data[3] = 6;
            }
            case "Juli" -> {
                data[2] = 0;
                data[3] = 7;
            }
            case "Agustus" -> {
                data[2] = 0;
                data[3] = 8;
            }
            case "September" -> {
                data[2] = 0;
                data[3] = 9;
            }
            case "Oktober" -> {
                data[2] = 1;
                data[3] = 0;
            }
            case "November" -> {
                data[2] = 1;
                data[3] = 1;
            }
            case "Desember" -> {
                data[2] = 1;
                data[3] = 2;
            }
            default -> {
            }
        }
        
        System.out.println("Inputkan Tahun = ");
        tahun = input.nextInt();
        
        for(int i = 7; i > 3; i--){
            data[i] = tahun % 10;
            tahun /= 10;
        }
        
        j = 7;
        for(int i = 0; i < 8; i++){
            if(data[i] != data[j]){
                temp += 1;
            }
            j -= 1;
        }
        
        if(temp == 0 ){
            System.out.println("Palindrome"); 
        }else{
            System.out.println("Not Palindrome");
        }
    }
    
}
